new_webapp/src/actions/index.js

```diff
@@ -3,13 +3,15 @@ import _ from "underscore";
 import { Search, ghciUsage, hooglePlusExampleSearch, hooglePlusMoreExamples } from "../gateways";
 import { namedArgsToUsage, getArgCount } from "../utilities/args";
 import { LOADING, DONE, ERROR } from "../constants/fetch-states";
+import { log } from "../utilities/logger";
 
 function makeActionCreator(type, ...argNames) {
     return function (...args) {
         const action = { type }
         argNames.forEach((arg, index) => {
             action[argNames[index]] = args[index]
         })
+		log.info(type,action)
```

Comment:
btw, the _right_ way to do this is here:
It's a walkthrough of how one starts at basically this implementation and ends at a more general solution.
https://redux.js.org/advanced/middleware/

You don't need to do that if you don't want to. But it's likely this will omit some action in the future that doesn't use the `makeActionCreator` function.